src/docfx/lib/git/FileCommitProvider.cs

```diff
@@ -34,27 +38,48 @@ internal sealed class FileCommitProvider : IDisposable
         // Commit history LRU cache per file. Key is the file path relative to repository root.
         // Value is a dictionary of git commit history for a particular commit hash and file blob hash.
         // Only the last N = MaxCommitCacheCountPerFile commit histories are cached for a file, they are selected by least recently used order (lruOrder).
-        private readonly ConcurrentDictionary<string, Dictionary<(long commit, long blob), (long[] commitHistory, int lruOrder)>> _commitCache;
+        private readonly Lazy<Task<ConcurrentDictionary<string, Dictionary<(long commit, long blob), (long[] commitHistory, int lruOrder)>>>> _commitCache;
 
         private int _nextLruOrder;
         private int _nextStringId;
         private bool _cacheUpdated;
         private IntPtr _repo;
 
-        public FileCommitProvider(
-            string repoPath,
-            ConcurrentDictionary<string, Dictionary<(long commit, long blob), (long[] commitHistory, int lruOrder)>> commitCache)
+        public FileCommitProvider(string repoPath, string cacheFilePath)
         {
             if (git_repository_open(out _repo, repoPath) != 0)
             {
                 throw new ArgumentException($"Invalid git repo {repoPath}");
             }
             _repoPath = repoPath;
+            _cacheFilePath = cacheFilePath;
             _commits = new ConcurrentDictionary<string, Lazy<(List<Commit>, Dictionary<long, Commit>)>>();
-            _commitCache = commitCache;
+            _commitCache = new Lazy<Task<ConcurrentDictionary<string, Dictionary<(long commit, long blob), (long[] commitHistory, int lruOrder)>>>>(
+                () => LoadCommitCache(_cacheFilePath));
+        }
+
+        public async Task<List<GitCommit>> GetCommitHistory(string file, string committish = null)
+        {
+            var commitCache = await _commitCache.Value;
+            var fileCommitCache = commitCache.GetOrAdd(file, _ => new Dictionary<(long, long), (long[], int)>());
+
+            return GetCommitHistory(file, int.MaxValue, committish, fileCommitCache);
+        }
+
+        public List<GitCommit> GetCommitHistory(string file, int top, string committish = null)
```

Comment:
normal file should not call this method ever, so why not change it to GetDeletedCommitHistory

Reply:
and no need leverage cache anyway

Reply:
as we talked offline, these two method should not have any relationship, so one is with cache, one is without cache